homeassistant/components/mysensors/config_flow.py

```diff
@@ -0,0 +1,229 @@
+"""Config flow for MySensors."""
+import logging
+from typing import Dict, Optional
+
+import voluptuous as vol
+
+from homeassistant.components.mysensors import (
+    CONF_DEVICE,
+    DEFAULT_BAUD_RATE,
+    DEFAULT_TCP_PORT,
+    is_persistence_file,
+)
+import homeassistant.helpers.config_validation as cv
+
+from . import CONF_RETAIN, CONF_VERSION, DEFAULT_VERSION
+from ... import config_entries
+from ..mqtt import valid_publish_topic, valid_subscribe_topic
+from .const import (
+    CONF_BAUD_RATE,
+    CONF_GATEWAY_TYPE,
+    CONF_GATEWAY_TYPE_ALL,
+    CONF_GATEWAY_TYPE_MQTT,
+    CONF_GATEWAY_TYPE_SERIAL,
+    CONF_GATEWAY_TYPE_TCP,
+    CONF_GATEWAY_TYPE_TYPE,
+    CONF_PERSISTENCE_FILE,
+    CONF_TCP_PORT,
+    CONF_TOPIC_IN_PREFIX,
+    CONF_TOPIC_OUT_PREFIX,
+    DOMAIN,
+)
+from .gateway import MQTT_COMPONENT, is_serial_port, is_socket_address, try_connect
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_schema_common() -> dict:
+    """Create a schema with options common to all gateway types."""
+    schema = {
+        vol.Required(
+            CONF_VERSION, default="", description={"suggested_value": DEFAULT_VERSION}
+        ): str,
+        vol.Optional(
+            CONF_PERSISTENCE_FILE,
+        ): str,
+    }
+    return schema
+
+
+def _validate_version(version: str) -> Dict[str, str]:
+    """Validate a version string from the user."""
+    errors = {CONF_VERSION: "invalid_version"}
+    version_parts = version.split(".")
```

Comment:
Can we parse the version with a version parser from eg `packaging` instead?

Reply:
Sounds like a good idea. In particular i would love to use `semver`. However, MySensors versions are not compliant to it, so we have to find something else.
The pymysensors library uses
```
# pylint: disable=no-name-in-module, import-error
from distutils.version import LooseVersion as parse_ver
```
I tried it out and found that LooseVersion really accepts anything. But there is also StrictVersion, which seems to work just fine.

Maybe we could use that? It would save us from adding another dependency.

Reply:
Please use this:
```py
from packaging.version import parse as parse_version
```